docs/developer.md

```diff
@@ -0,0 +1,99 @@
+# DEVELOPER GUIDE --- WIP
+
+## Contributing
+// TODO(user): Add detailed information on how you would like others to contribute to this project
+
+### How it works
+This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).
+
+It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/)
+which provides a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
+
+
+## Local development
+
+### Prerequisites
+
+You need ready to use **kubectl** and **minikube** [tools](https://kubernetes.io/docs/tasks/tools/) as a prerequisite.
+
+Get your copy of Backstage Operator from GitHub:
+```sh
+git clone https://github.com/janus-idp/operator
+```
+
+### Local Tests
+
+Including unit testing (starting from v 0.02) and Kubernetes intergation testing ([envtest](https://book.kubebuilder.io/reference/envtest.html))
```

Comment:
```suggestion
To run both unit tests (@since 0.0.2) and Kubernetes integration tests ([envtest](https://book.kubebuilder.io/reference/envtest.html)): 
```